File: endpoints/viewphotocategory.py
from silence.decorators import endpoint

@endpoint(
    route="/viewphotocategory",
    method="GET",
    sql="SELECT * FROM viewphotocategory"
)
def get_all():
    pass

###############################################################################

@endpoint(
    route="/viewphotocategory/$categoryId",
    method="GET",
    sql="SELECT * FROM viewphotocategory WHERE categoryId = $categoryId"
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/viewphotocategory/photo/$photoId",
    method="GET",
    sql="SELECT * FROM viewphotocategory WHERE photoId = $photoId"
)
def get_category_by_id():
    pass

File: docs/js/api/viewphotocategory.js
/*
	C.Arévalo
	photos.js.  API Rest para recurso:"photos"
	Marzo/2021
*/
"use_strict";
import { BASE_URL , requestOptions } from "./common.js";
const viewPhotoCategoryAPI = {
    
    getAll: function () {
        return new Promise ( function ( resolve , reject ) {
            axios
            .get(`${BASE_URL}/viewphotocategory`, requestOptions )
            .then ( response => resolve ( response.data ))
            .catch ( error => reject ( error.response.data.message ));
            });
        },
    getById: function ( categoryId ) {
        return new Promise ( function ( resolve , reject ) {
        axios
        .get(`${BASE_URL}/viewphotocategory/${categoryId}`, requestOptions )
        .then( response => resolve ( response.data ))
        .catch( error => reject ( error.response.data.message ));
        });
    },
    getCategoryById: function ( photoId ) {
        return new Promise ( function ( resolve , reject ) {
        axios
        .get(`${BASE_URL}/viewphotocategory/photo/${photoId}`, requestOptions )
        .then( response => resolve ( response.data ))
        .catch( error => reject ( error.response.data.message ));
        });
    },
    
       
};


export { viewPhotoCategoryAPI };